### PR TITLE
External UCS parameters: Script 001

### PR DESCRIPTION
**Pull request by dboltovskyi**
Script is part of test set: https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1169
Common functions for CRQ: https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1184

### CODE REVIEW

File: test_scripts/Polices/External_UCS/001_ATF_P_TC_LPT_Creation_PreloadedPT_without_external_consent_status_groups_struct.lua [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+--[[ Test ]]
+  commonFunctions:newTestCasesGroup("Test")
+
+  function Test.StartSDL()
+    StartSDL(config.pathToSDL, config.ExitOnCrash)
+    os.execute("sleep 1")
+  end
+
+  function Test:CheckSDLStatus()
```
Comment (istoimenova) [inline test_scripts/Polices/External_UCS/001_ATF_P_TC_LPT_Creation_PreloadedPT_without_external_consent_status_groups_struct.lua]: @dboltovskyi, please add some delay after starting SDL, as usual SDL needs time to check preloaded_pt.json file and in this case SDL can stop after verification is done.

File: test_scripts/Polices/External_UCS/001_ATF_P_TC_LPT_Creation_PreloadedPT_without_external_consent_status_groups_struct.lua [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+--[[ General Settings for configuration ]]
+  Test = require("user_modules/connecttest_resumption")
+  require('user_modules/AppTypes')
+
+--[[ Preconditions ]]
+  commonFunctions:newTestCasesGroup("Preconditions")
+
+  function Test:CheckSDLStatus()
+    testCasesForExternalUCS.checkSDLStatus(self, sdl.RUNNING, "SDL is not running")
```
Comment (dboltovskyi) [inline test_scripts/Polices/External_UCS/001_ATF_P_TC_LPT_Creation_PreloadedPT_without_external_consent_status_groups_struct.lua]: @istoimenova, It was a mistake in description. Correct status is RUNNING

File: test_scripts/Polices/External_UCS/001_ATF_P_TC_LPT_Creation_PreloadedPT_without_external_consent_status_groups_struct.lua [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  end
+
+  function Test:StopSDL()
```
Comment (dboltovskyi) [inline test_scripts/Polices/External_UCS/001_ATF_P_TC_LPT_Creation_PreloadedPT_without_external_consent_status_groups_struct.lua]: @istoimenova, The idea here was to make sure SDL works correctly before modification of PreloadedPT

File: test_scripts/Polices/External_UCS/001_ATF_P_TC_LPT_Creation_PreloadedPT_without_external_consent_status_groups_struct.lua [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  end
+
+  function Test.RemoveLPT()
+    testCasesForExternalUCS.removeLPT()
+  end
+
+  function Test.UpdatePreloadedPT()
+    local updateFunc = function(preloadedTable)
+      -- whole 'device_data' section is not allowed in PreloadedPT according to dictionary
+      preloadedTable.policy_table.device_data = nil
```
Comment (dboltovskyi) [inline test_scripts/Polices/External_UCS/001_ATF_P_TC_LPT_Creation_PreloadedPT_without_external_consent_status_groups_struct.lua]: @istoimenova, Yes it is not observed now. 
I've just added it in order to have more clear picture and to conform req-t
If we remove this part req-t becomes useless. There is no modification in PreloadedPT. So we will check that SDL works correctly with the default one. But SDL should do this.

File: test_scripts/Polices/External_UCS/001_ATF_P_TC_LPT_Creation_PreloadedPT_without_external_consent_status_groups_struct.lua [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    end
+    testCasesForExternalUCS.updatePreloadedPT(updateFunc)
```
Comment (dboltovskyi) [inline test_scripts/Polices/External_UCS/001_ATF_P_TC_LPT_Creation_PreloadedPT_without_external_consent_status_groups_struct.lua]: @istoimenova, According to data dictionary whole section 'device_data' shouldn't be defined in PreloadedPT in order SDL starts successfully. That's why the whole section is removed in line 68

File: test_scripts/Polices/External_UCS/001_ATF_P_TC_LPT_Creation_PreloadedPT_without_external_consent_status_groups_struct.lua (hunk 1 above)
Comment (dboltovskyi) [inline test_scripts/Polices/External_UCS/001_ATF_P_TC_LPT_Creation_PreloadedPT_without_external_consent_status_groups_struct.lua]: @istoimenova, Delay for 1 sec is defined in line 78

File: test_scripts/Polices/External_UCS/001_ATF_P_TC_LPT_Creation_PreloadedPT_without_external_consent_status_groups_struct.lua [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+-- Preconditions:
+-- 1. Stop SDL
+-- 2. Modify PreloadedPolicyTable (remove 'external_consent_status_groups' section)
+--
+-- Steps:
+-- 1. Start SDL
+-- 2. Check SDL status
+--
+-- Expected result:
+-- Status = 0 (SDL is stopped)
```
Comment (dboltovskyi) [inline test_scripts/Polices/External_UCS/001_ATF_P_TC_LPT_Creation_PreloadedPT_without_external_consent_status_groups_struct.lua]: @istoimenova, There was a mistake in description. It was corrected. Please check cd04ff1

File: test_scripts/Polices/External_UCS/001_ATF_P_TC_LPT_Creation_PreloadedPT_without_external_consent_status_groups_struct.lua [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+--[[ Required Shared Libraries ]]
+  local commonFunctions = require('user_modules/shared_testcases/commonFunctions')
+  local commonSteps = require('user_modules/shared_testcases/commonSteps')
+  local commonPreconditions = require('user_modules/shared_testcases/commonPreconditions')
+  local sdl = require('SDL')
+  local testCasesForExternalUCS = require('user_modules/shared_testcases/testCasesForExternalUCS')
+
+--[[ General Precondition before ATF start ]]
+  commonFunctions:SDLForceStop()
+  commonSteps:DeleteLogsFileAndPolicyTable()
```
Comment (dboltovskyi) [inline test_scripts/Polices/External_UCS/001_ATF_P_TC_LPT_Creation_PreloadedPT_without_external_consent_status_groups_struct.lua]: @istoimenova, Corrected, please check cd04ff1

File: test_scripts/Polices/External_UCS/001_ATF_P_TC_LPT_Creation_PreloadedPT_without_external_consent_status_groups_struct.lua (hunk 4 above)
Comment (istoimenova) [inline test_scripts/Polices/External_UCS/001_ATF_P_TC_LPT_Creation_PreloadedPT_without_external_consent_status_groups_struct.lua]: @dboltovskyi, there is non-conformity between this and upper comment.
In previous you say that wants to check SDL behaviour before update of PreloadedPT, in this comment you say that PreloadedPT will be not updated.
Please clarify.

File: test_scripts/Polices/External_UCS/001_ATF_P_TC_LPT_Creation_PreloadedPT_without_external_consent_status_groups_struct.lua (hunk 5 above)
Comment (istoimenova) [inline test_scripts/Polices/External_UCS/001_ATF_P_TC_LPT_Creation_PreloadedPT_without_external_consent_status_groups_struct.lua]: @dboltovskyi, agree.

File: test_scripts/Polices/External_UCS/001_ATF_P_TC_LPT_Creation_PreloadedPT_without_external_consent_status_groups_struct.lua (hunk 1 above)
Comment (istoimenova) [inline test_scripts/Polices/External_UCS/001_ATF_P_TC_LPT_Creation_PreloadedPT_without_external_consent_status_groups_struct.lua]: @dboltovskyi, 
suggestion: according to me 5 sec will be better to be wait to check SDL status.

**Comment by dboltovskyi** [comment]
@istoimenova, 

> @dboltovskyi, there is non-conformity between this and upper comment.
In previous you say that wants to check SDL behaviour before update of PreloadedPT, in this comment you say that PreloadedPT will be not updated.
Please clarify.

There is no confusion in my words.
Standard way to verify something consists of the following steps:
1. Check that system works before modification
2. Do the changes
3. Check that system works as expected
So script is correspond to these 3 steps.

We have the following restrictions:
1) Section 'external_consent_status_groups' is located under the 'device_data' section which has to be omitted in PreloadedPT (according to data dictionary).
So removing (or adding) 'external_consent_status_groups' we have to remove (or add) 'device_data' section
2) By default section 'device_data' is omitted in PreloadedPT in order to SDL is able to run correctly

So bearing in mind all this information the only way to verify req-t is:
1. Modify PreloadedPT by adding 'external_consent_status_groups' along with 'device_data' section
2. Try to start SDL => get 0 status (STOPPED) due to defined 'device_data' section
3. Remove 'external_consent_status_groups' along with 'device_data' section
4. Try to start SDL again => get 1 status (RUNNING)

What do you think is it a right way?
If so script has to be corrected

**Comment by dboltovskyi** [comment]
@istoimenova,
> suggestion: according to me 5 sec will be better to be wait to check SDL status.

Current version (1 sec) works fine - I did a lot of runs while debugging the scripts

**Pull request merged**